base/iam.tf

```diff
@@ -0,0 +1,186 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_iam_role" "external_dns_role" {
+  name = "${var.name}_external_dns_role"
+  path = "/kube2iam/"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "AWS": [
+          "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.cluster_nodes_iam_role_name}",
+          "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.cluster_masters_iam_role_name}"
+        ]
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "external_dns_role_policy" {
+  name = "${var.name}_external_dns_policy"
+  role = "${aws_iam_role.external_dns_role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "route53:*"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+data "aws_region" "fluentd_region" {}
```

Comment:
You have another `aws_region` data provider to get the current region, line 5 of the `main.tf` file:

```
data "aws_region" "current" {}
```

Reply:
Right, missed that one, I'll make sure we've got only one 😓